### ISSUE TITLE
Fix S4143 FP: Indexers

### ISSUE THREAD

**Issue by przemyslaw-piescik** [SonarSource/sonar-dotnet#4178]
### Description

S4143 False positive when using indexers.

### Repro steps

```cs
var first = new Curve();
var second = new Curve();

first.Points[0] = new Point();
second.Points[0] = new Point();
```
S4143 is reported, although the index in accessed on separate objects.

### Expected behavior

No warning reported.

### Actual behavior

S4143 is reported.

**Comment by zamojski** [comment]
I've very similar issue, when can I expect this one to get fixed?
